File: express-ts/src/modules/user/user.repository.ts
export type User = {
  id: number;
  username: string;
  name: string;
  email: string;
  password: string;
}

export interface UserRepositoryInterface {
  getUserByID(id: number): User | undefined;
  getUserByEmail(email: string): User | undefined;
}

class UserRepository implements UserRepositoryInterface {
  private users: User[] = [{
    id: 1,
    username: 'testuser',
    name: 'Test User',
    email: 'user1@test.com',
    password: 'passworduser1'
  }];

  getUserByID(id: number): User | undefined {
    return this.users.find(user => user.id === id);
  }

  getUserByEmail(email: string): User | undefined {
    return this.users.find(user => user.email.toLowerCase() === email.toLowerCase());
  }
}

export const userRepository = new UserRepository();

File: javalin/src/main/java/com/jwt/app/user/repository/UserRepository.java
package com.jwt.app.user.repository;

import com.jwt.app.user.dto.User;

import java.util.ArrayList;
import java.util.List;

public class UserRepository {
    private final List<User> users = new ArrayList<>();

    public UserRepository() {
        users.add(new User(1, "testuser", "Test User", "user1@test.com", "passworduser1"));
    }

    public User findById(int id) {
        return users.stream().filter(user -> user.getId() == id).findFirst().orElse(null);
    }

    public User findByEmail(String email) {
        return users.stream().filter(user -> user.getEmail().equalsIgnoreCase(email)).findFirst().orElse(null);
    }
}